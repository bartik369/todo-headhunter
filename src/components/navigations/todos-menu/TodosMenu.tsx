import {FC, useEffect, useState} from 'react';
import {useAppDispatch, useAppSelector } from '../../../hooks/useReduxHook';
import { sortTodos, deleteComplitedTodos} from '../../../store/TodoSlice';
import { itemValuesArr } from '../../../utils/menuData';
import { useCountHook } from '../../../hooks/useCountHook';
import { todosMenuData } from '../../../utils/menuData';
import * as todoConst from '../../../constants/todos';
import style from './TodosMenu.module.scss';

const TodosMenu:FC = () => {
    const [activeMenu, setActiveMenu] = useState(1);
    const todos = useAppSelector(state => state?.todos?.todos);
    const [countItems, setCountItems] = useState(0);
    const dispatch = useAppDispatch();
    const [getItemValues] = useCountHook();

    const handleMenu = (id: number) => {
        dispatch(sortTodos(id));
        setActiveMenu(id);
    };
    const resetHandler = () => {
        dispatch(deleteComplitedTodos());
    };

    useEffect(() => {
        dispatch(sortTodos(activeMenu));
    }, [resetHandler]);

    useEffect(() => {
        const counts = todos?.filter(item => !item.status);
        setCountItems(counts?.length);
    }, [todos]);
    
    return (
      <div className={style.panel}>
        <div className={style.total}>
          {getItemValues(countItems, itemValuesArr)}
          {todoConst.items}
        </div>
        <div className={style.navigation}>
        <menu className={style.menu}>
          <ul>
            {todosMenuData.map((item) => (
              <li
                className={activeMenu === item.id ? style.selected : ""}
                onClick={() => handleMenu(item.id)}
                key={item.id}
              >
                {item.title}
              </li>
            ))}
          </ul>
        </menu>
        <button className={style.reset} onClick={resetHandler}>
          {todoConst.resetComplited}
        </button>
        </div>
      </div>
    );
};

export default TodosMenu;