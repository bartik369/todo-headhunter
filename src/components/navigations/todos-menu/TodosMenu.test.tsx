import { render } from "@testing-library/react";
import TodosMenu from "./TodosMenu";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe('Test TodosMenu component', () => {
  const initialState = {
    todos: []
  };
  const mockStore = configureStore();
  let store= mockStore(initialState);

  test('Render', () => {
    render (
      <Provider store={store}>
        <TodosMenu />
      </Provider>
    );
  });
  test('Snapshot', () => {
    const menuList = render(
      <Provider store={store}>
        <TodosMenu />
      </Provider>
    )
    expect(menuList).toMatchSnapshot();
  })
});