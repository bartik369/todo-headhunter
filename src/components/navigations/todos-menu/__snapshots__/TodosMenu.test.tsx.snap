// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodosMenu component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="panel"
      >
        <div
          class="total"
        >
          undefined items
           left
        </div>
        <menu
          class="menu"
        >
          <ul>
            <li
              class="selected"
            >
              All
            </li>
            <li
              class=""
            >
              Active
            </li>
            <li
              class=""
            >
              Completed
            </li>
          </ul>
        </menu>
        <button
          class="reset"
        >
          Clear completed
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="panel"
    >
      <div
        class="total"
      >
        undefined items
         left
      </div>
      <menu
        class="menu"
      >
        <ul>
          <li
            class="selected"
          >
            All
          </li>
          <li
            class=""
          >
            Active
          </li>
          <li
            class=""
          >
            Completed
          </li>
        </ul>
      </menu>
      <button
        class="reset"
      >
        Clear completed
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
